src/core/credentials/transport/tls/grpc_tls_credentials_options.h

```diff
@@ -88,6 +90,10 @@ struct grpc_tls_credentials_options
   void set_crl_directory(std::string crl_directory) { crl_directory_ = std::move(crl_directory); }
   void set_crl_provider(std::shared_ptr<grpc_core::experimental::CrlProvider> crl_provider) { crl_provider_ = std::move(crl_provider); }
   void set_send_client_ca_list(bool send_client_ca_list) { send_client_ca_list_ = send_client_ca_list; }
+  // Sets the `grpc_tls_certificate_provider` to provide identity data.
+  void set_identity_certificate_provider(grpc_core::RefCountedPtr<grpc_tls_certificate_provider> certificate_provider)  { identity_cert_provider_ = std::move(certificate_provider); }
```

Comment:
How much work would it be to modify this to use `std::shared_ptr` instead of `RefCountedPtr` at just the API layer?

`RefCountedPtr` is a custom type we used to have to use because we could use smart pointers, but now we can